src/Uno.UI/UI/Xaml/ResourceDictionary.cs

```diff
@@ -424,6 +425,11 @@ internal void CopyFrom(ResourceDictionary source)
 			_mergedDictionaries.Clear();
 			_themeDictionaries?.Clear();
 
+			// In a foreign library that uses merged-dict with 'Source=...', there can be multiple instances of res-dict from one single xaml file.
+			// And, the instance referenced in App.xaml may not be the same one used in merged-dict of library-a\textbox.xaml.
+			// In order to ensure the theme updates covers all of them, we need to keep track of this source instance for theme updates before it is lost.
+			_sourceDictionary = new WeakReference<ResourceDictionary>(source);
```

Comment:
Use `WeakReferencePool.Rent()` instead, it'll reuse the self-defined weak reference in `ResourceDictionary`